include/deal.II/base/mpi.h

```diff
@@ -1193,70 +1193,36 @@ namespace Utilities
             const unsigned int    root_process = 0);
 
     /**
-     * This function sends an object @p object_to_send from the process @p root_process
-     * to all other processes.
+     * This function sends an object @p object_to_send from the process @p
+     * root_process to all other processes.
      *
      * This function is a generalization of the classic `MPI_Bcast` function
-     * that accepts arbitrary data types `T`, as long as Utilities::pack()
-     * (which in turn uses `boost::serialize`, see in Utilities::pack() for
-     * details) accepts `T` as an argument.
-     *
-     * @note Be aware that this function is typically a lot more
-     * expensive than an `MPI_Bcast` because the function will use
-     * boost::serialization to (de)serialize, and execute a second
-     * `MPI_Bcast` to transmit the size before sending the data
-     * itself. On the other hand, if you have a single element of
-     * a data type `T` that is natively supported by MPI, then the
-     * compiler will choose another broadcast() overload that is efficient.
-     * If you have an array of such elements, you should use the other
-     * broadcast() function in this namespace that takes a pointer and a count
-     * argument.
-     *
-     * @param[in] comm MPI communicator.
-     * @param[in] object_to_send An object to send to all processes.
-     * @param[in] root_process The process that sends the object to all
-     * processes. By default the process with rank 0 is the root process.
-     *
-     * @tparam T Any type for which the Utilities::pack() and
-     *   Utilities::unpack() functions can be used to convert the object
-     *   into an array of `char`. The compiler will not select this function
-     *   if `T` is a type that is natively supported by MPI and instead
-     *   use a more efficient overload.
-     *
-     * @return On the root process, return a copy of @p object_to_send.
-     *   On every other process, return a copy of the object sent by
-     *   the @p root_process.
-     */
-    template <typename T>
-    std::enable_if_t<is_mpi_type<T> == false, T>
-    broadcast(const MPI_Comm     comm,
-              const T           &object_to_send,
-              const unsigned int root_process = 0);
-
-    /**
-     * This function sends an object @p object_to_send from the process @p root_process
-     * to all other processes.
+     * that accepts arbitrary data types `T`. If `T` is an MPI type then this
+     * function immediately calls `MPI_Bcast`. Otherwise it will first use
+     * Utilities::pack() (which in turn uses `boost::serialize`, see in
```

Comment:
```suggestion
     * Utilities::pack() (which in turn uses `boost::serialize`, see
```

Reply:
I combined this with the former note (now a warning)